### PR TITLE
Script + Workflow for resizing cover image thumbnails

### PR DESCRIPTION
**Pull request by codemonkey800**
## Description

Adds script for resizing cover images into smaller thumbnails for loading on the collection page. This reduces the average uncached load time from 1s - 15s to  300ms - 600ms

I also added a workflow that automatically resizes images and commits the resized images to the current PR if the author adds or modifies an image file

### CODE REVIEW

File: images/resize.mjs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+const SUPPORTED_IMAGE_EXTENSIONS = ['jpeg', 'jpg', 'png']
+
+const imageFiles = fs
+  .readdirSync('.')
+  .filter(
+    file =>
+      !file.includes('-thumb') &&
+      SUPPORTED_IMAGE_EXTENSIONS.some(ext => file.includes(ext))
+  )
```
Comment (manasaV3) [inline images/resize.mjs:14]: Could we also add a filter to remove images that already have a thumbnail?

File: images/resize.mjs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  console.log(`Resizing ${imageFiles.length} images`)
+
+  imageFiles.forEach(async file => {
+    const image = sharp(file)
+    const metadata = await image.metadata()
+
+    if ((metadata.width ?? 0) > THUMB_IMAGE_WIDTH) {
+      image.resize(THUMB_IMAGE_WIDTH).toFile(getThumbFile(file), err => {
+        if (err) {
+          console.error(`Unable to resize image`, err)
```
Comment (manasaV3) [inline images/resize.mjs:38]: Would the err include the file name?

File: images/resize.mjs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  const ext = path.extname(file)
+  const name = path.basename(file, ext)
+  return `${name}-thumb${ext}`
+}
+
+/**
+ * The max observed width for images in the browser. This happens between 600px
+ * and 875px when the page is using a 2 column flexible layout.
+ */
+const THUMB_IMAGE_WIDTH = 500
```
Comment (klai95) [inline images/resize.mjs:26]: How did we decide on this width?

File: images/resize.mjs (hunk 1 above)
Comment (manasaV3) [reply images/resize.mjs]: Ahh... That makes sense that we could update the actual images.. In that case, I think retaining it as it is right now, would be lesser hassle..

**Review by manasaV3 — APPROVED** [review]

**Pull request merged**